fw/thread.go

```diff
@@ -298,22 +299,38 @@ func (t *Thread) processIncomingInterest(packet *defn.Pkt) {
 	if packet.NextHopFaceID != nil {
 		if dispatch.GetFace(*packet.NextHopFaceID) != nil {
 			core.LogTrace(t, "NextHopFaceId is set for Interest ", packet.Name, " - dispatching directly to face")
-			dispatch.GetFace(*packet.NextHopFaceID).SendPacket(packet)
+			dispatch.GetFace(*packet.NextHopFaceID).SendPacket(dispatch.OutPkt{
+				Pkt:      packet,
+				PitToken: packet.PitToken, // TODO: ??
+				InFace:   packet.IncomingFaceID,
+			})
 		} else {
 			core.LogInfo(t, "Non-existent face specified in NextHopFaceId for Interest ", packet.Name, " - DROP")
 		}
 		return
 	}
 
-	// Pass to strategy AfterReceiveInterest pipeline
-	var nexthops []*table.FibNextHopEntry
-	if fhName == nil {
-		nexthops = table.FibStrategyTable.FindNextHopsEnc(interest.NameV)
-	} else {
-		nexthops = table.FibStrategyTable.FindNextHopsEnc(fhName)
+	// Use forwarding hint if present
+	lookupName := interest.NameV
+	if fhName != nil {
+		lookupName = fhName
 	}
 
-	strategy.AfterReceiveInterest(packet, pitEntry, incomingFace.FaceID(), nexthops)
+	// Query the FIB for possible nexthops
+	nexthops := table.FibStrategyTable.FindNextHopsEnc(lookupName)
+
+	// Exclude faces that have an in-record for this interest
+	// TODO: unclear where NFD dev guide specifies such behavior (if any)
```

Comment:
I guess that's to prevent loop: an incoming face indicates that path does not have the desired Data.

Reply:
Yeah, but I didn't see this condition mentioned in the dev guide .. seems kinda obvious, maybe I missed it ...

Reply:
FYI, the NFD dev guide is not intended as a full specification of all NDN forwarding rules, it's supposed to "just" describe the implementation in NFD. I lost count of how many times we tried (and never finished) to write a generic and implementation-agnostic spec of the NDN forwarding behavior 😞